### PR TITLE
Fixes #21795: Add security headers in Lift config

### PR DESCRIPTION
**Pull request by amousset**
https://issues.rudder.io/issues/21795

### CODE REVIEW

**Comment by amousset** [comment]
PR updated with a new commit

File: webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/Boot.scala [hunk 1]
```diff
@@ -244,7 +244,7 @@ class Boot extends Loggable {
 
     //exclude Rudder doc from context-path rewriting
     LiftRules.excludePathFromContextPathRewriting.default.set(() => (path:String) => {
-      val noRedirectPaths= "/rudder-doc" :: "/ncf" :: "/ncf-builder" :: Nil
+      val noRedirectPaths= "/rudder-doc" :: Nil
```
Comment (fanf) [inline webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/Boot.scala:247]: I think that's a bug and should be removed for all 7.x branches

**Review by fanf — APPROVED** [review]
Seems good, but we will need to test for real. I don't know if we can unit test that with lift test facilities

File: webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/Boot.scala (hunk 1 above)
Comment (amousset) [reply webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/Boot.scala:247]: It is a leftover that does not exist anymore, not really a bug per se IMHO

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
@fanf added a setting for HSTS as it is not always safe to enable.

**Comment by amousset** [comment]
I still need to be able to test these.

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Pull request merged**